src/rospkg/rospack.py

```diff
@@ -89,9 +90,15 @@ def list_by_path(manifest_name, path, cache):
             # optimization for stacks.
             del dirs[:]
             continue #leaf     
-        elif 'rospack_nosubdirs' in files:
+        elif 'rospack_nosubdirs' in files or not recurse:
             del dirs[:]
             continue  #leaf
+        elif 'rospack_norecurse' in files:
```

Comment:
I'm fine with the concept of `rospack_norecurse`. I don't think we have to resolve how the file gets placed in the needed directories, e.g. `/usr/share`, to agree that it could be useful just like `rospack_nosubdirs` or `CATKIN_IGNORE`.

However, for this to be merged, I agree with @dirk-thomas that it needs to be documented and other tools that currently respect `rospack_nosubdirs` also need to be updated. At least `rospack` is in that list, possibly others, though tools should use either `rospkg` (Python) or `rospack` (C++) api's rather than implement this themselves, so the list should be short.